generator/src/main/java/com/scottlogic/deg/generator/walker/ReductiveWalkerRetryChecker.java

```diff
@@ -2,26 +2,26 @@
 
 public class ReductiveWalkerRetryChecker {
     private int numRetriesSoFar = 0;
-    private boolean guaranteedNotFullyContradictory = false;
+    private boolean successOccurredAtLeastOnce = false;
     private int retryLimit;
 
     public ReductiveWalkerRetryChecker(int retryLimit) {
         this.retryLimit = retryLimit;
     }
 
-    void setRetrySuccessful() {
-        guaranteedNotFullyContradictory = true;
+    void retrySuccessful() {
+        successOccurredAtLeastOnce = true;
     }
 
-    void setRetryUnsuccessful() {
+    void retryUnsuccessful() {
         numRetriesSoFar++;
-        if (numRetriesSoFar > retryLimit && !guaranteedNotFullyContradictory) {
+        if (numRetriesSoFar > retryLimit && !successOccurredAtLeastOnce) {
```

Comment:
`!successOccurredAtLeastOnce` could be extracted out higher, as we no longer care about incrementing the `numTriesSoFar` once we have found at least one value.